service/kernel/env_detect.cpp

```diff
@@ -137,13 +137,13 @@ DAAL_EXPORT void daal::services::Environment::initNumberOfThreads()
 {
     if (isInit) return;
 
-    /* if HT enabled - set _numThreads to physical cores num */
+    /* if HT enabled - set number of threads to physical cores num */
     if (daal::internal::Service<>::serv_get_ht())
     {
         /* Number of cores = number of cpu packages * number of cores per cpu package */
-        int ncores = daal::internal::Service<>::serv_get_ncpus() * daal::internal::Service<>::serv_get_ncorespercpu();
+        const size_t ncores = daal::internal::Service<>::serv_get_ncpus() * daal::internal::Service<>::serv_get_ncorespercpu();
```

Comment:
SDL violation.

Reply:
This change was reverted and added casting to size_t for ncores in setNumberOfThreads() call